### PR TITLE
feat: Initial SQL support

### PR DESCRIPTION
**Pull request by tjholm**
Add support for collecting SQL database requirements and building migrations.

### CODE REVIEW

**Review by davemooreuws — DISMISSED** [review]
Works great. Only thing is `nitric start` does not show any progress when pulling postgres locally.

**Comment by tjholm** [comment]
> Works great. Only thing is `nitric start` does not show any progress when pulling postgres locally.

Have implemented a quick version of this with a simple `Starting Local Cloud` message.

**Review by davemooreuws — DISMISSED** [review]

File: pkg/cloud/sql/sql.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+		},
+		PortBindings: map[nat.Port][]nat.PortBinding{
+			// TODO: Randomize port number to allow multiple starts
+			"5432/tcp": {
+				{
+					HostPort: fmt.Sprint(freeport),
+				},
+			},
+		},
+		// TODO: Randomize instance name to allow multiple starts
```
Comment (davemooreuws) [inline pkg/cloud/sql/sql.go]:
```suggestion
```

File: pkg/cloud/sql/sql.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+		// },
+		Mounts: []mount.Mount{
+			{
+				Type:   mount.TypeVolume,
+				Source: volume.Name,
+				Target: "/var/lib/postgresql/data",
+			},
+		},
+		PortBindings: map[nat.Port][]nat.PortBinding{
+			// TODO: Randomize port number to allow multiple starts
```
Comment (davemooreuws) [inline pkg/cloud/sql/sql.go]:
```suggestion
```

**Review by davemooreuws — CHANGES_REQUESTED** [review]

**Review by davemooreuws — APPROVED** [review]

**Review by HomelessDinosaur — APPROVED** [review]

**Pull request merged**